torch/csrc/cuda/expand_utils.cpp

```diff
@@ -0,0 +1,188 @@
+#include "torch/csrc/cuda/THCP.h"
```

Comment:
Can we maybe create a generic dir in utils and generate these specializations via macros? It's going to be much less repetetive